src/main/java/com/project/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.project.cinema.security;
+
+import com.project.cinema.exceptions.AuthenticationException;
+import com.project.cinema.lib.Inject;
+import com.project.cinema.lib.Service;
+import com.project.cinema.model.User;
+import com.project.cinema.service.UserService;
+import com.project.cinema.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User userFromDB = userService.findByEmail(email)
```

Comment:
let's make service returns user or null

Reply:
In which case it should return null? 
If user isn't found or password is incorrect it throws an custom exception, is it enough?

Reply:
your Dao layer returns Optional. UserService returns user or null. And here you check  if(user != null &&  userFromDB.getPassword().equals(hashedPassword)) return user

Reply:
true, thanks for explanation. fixed